### PR TITLE
Slurm rest api connection

### PR DESCRIPTION
**Pull request by rlkennedyreid**

### CODE REVIEW

File: files/supervisord.conf [hunk 1]
```diff
@@ -90,7 +90,9 @@ stderr_logfile_backups=5
 priority=100
 
 [program:slurmrestd]
-user=slurmrestd
+# Must be executed as root to open sockets - but user will be changed with setuid()
+# https://slurm.schedmd.com/slurmrestd.html#lbAE - see '-u' and '-g' flags
+user=root
```
Comment (drewsilcock) [inline files/supervisord.conf:95]: Good spot

File: Dockerfile [hunk 2]
```diff
@@ -89,39 +89,44 @@ RUN set -ex \
...
     && make install \
     && install -D -m644 etc/cgroup.conf.example /etc/slurm/cgroup.conf.example \
     && install -D -m644 etc/slurm.conf.example /etc/slurm/slurm.conf.example \
     && install -D -m600 etc/slurmdbd.conf.example /etc/slurm/slurmdbd.conf.example \
     && install -D -m644 contribs/slurm_completion_help/slurm_completion.sh /etc/profile.d/slurm_completion.sh \
     && popd \
-    && rm -rf slurm \
+    && rm -rf slurm
+
+RUN set -ex \
```
Comment (drewsilcock) [inline Dockerfile:102]: Does this need to be run as separate `RUN` command and is the `set -ex` important?

Comment (rlkennedyreid) [reply Dockerfile:102]: Doesn't need to be - but it allows the changing of these users or directories without needing to do a full rebuild. I think there'll be some moving around of this stuff in the near future too, so thought it best to stay as another layer.

The flags are pretty typical in dockerfiles, I've mainly put them in to match the RUN command above. -e is for quick fail on any non-zero exit code, -x just prints commands I believe.

Comment (drewsilcock) [reply Dockerfile:102]: Makes sense.

**Comment by drewsilcock** [comment]
Whitespace changes might make it difficult to pull changes from upstream but other than that looks good.

**Review by drewsilcock — APPROVED** [review]

**Pull request merged**